pyzeebe/task/task_builder.py

```diff
@@ -0,0 +1,76 @@
+import inspect
+import logging
+from typing import List, Callable, Dict, Tuple
+
+from pyzeebe import Job, TaskDecorator
+from pyzeebe.task.task import Task
+from pyzeebe.task.task_config import TaskConfig
+from pyzeebe.task.types import DecoratorRunner, JobHandler
+
+logger = logging.getLogger(__name__)
+
+
+def build_task(task_function: Callable, task_config: TaskConfig) -> Task:
+    if not task_config.variables_to_fetch:
+        task_config.variables_to_fetch = get_parameters_from_function(task_function)
+
+    if task_config.single_value:
+        task_function = convert_to_dict_function(task_function, task_config.variable_name)
```

Comment:
`task_config.variable_name` can potentially be `None` (?)

Reply:
It can only be `None` if `single_value` is false. I do feel that this may be a sign of bad design.

Reply:
It's the linter that's warning me (I'm using Pyright/Pylance in the editor, but mypy also complains).

mypy output: 

```
pyzeebe/task/task_builder.py:18: error: Argument 2 to "convert_to_dict_function" has incompatible type "Optional[str]"; expected "str"
```

This pleases the linter:

```py
if not task_config.variable_name:
    raise RuntimeError("task_config.variable_name not set")
```

Raising a RuntimeError is fair here, it should never happen, but who knows. 

I see that we have mypy as a project dependency, but we don't run it in CI. Will create a task for this.